tests/func/test_api.py

```diff
@@ -143,3 +146,11 @@ def _set_remote_url_and_commit(repo, remote_url):
     rconfig.modify("upstream", "url", remote_url)
     repo.scm.add([repo.config.config_file])
     repo.scm.commit("modify remote")
+
+
+def test_open_scm_controlled(dvc_repo, repo_dir):
```

Comment:
I would say one of the most important use cases for this is opening files that are not cached by us. E.g. `--metric-no-cache`. Need to test that one too.

Reply:
Hmm, but from the tool point of view, such metric does not differ from `*.dvc` file right?

Reply:
@pared from implementation? Well, yeah, but they might be tracked by git _or_ they might not be tracked by anything and just missing.

Reply:
and I think that they will be found by find_outs_by_path in _open, that is why I've commented https://github.com/iterative/dvc/pull/2566/files#r330900006 . So it seems like that is broken right now.

Reply:
Added 2 test cases.